File: bin/reverse_proxy.dart
#!/usr/bin/env dart

import 'dart:io';
import 'dart:isolate';
import 'package:where/where.dart';

/// Application entry point.
Future<void> main(List<String> args) async {
  final executable = await where('node', onError: (command) => command);
  final packageUri = await Isolate.resolvePackageUri(Uri.parse('package:reverse_proxy/'));
  final script = packageUri.resolve('../bin/reverse_proxy.js').toFilePath();
  return Process.start(executable, [script]..addAll(args), mode: ProcessStartMode.inheritStdio);
}

File: bin/reverse_proxy.js
#!/usr/bin/env node
import cluster from "cluster";
import {main} from "../lib/cli/main.js";

/**
 * Application entry point.
 * @return {Promise} Completes when the program is terminated.
 */
async function main() {
	const id = cluster.isMaster ? "master" : `worker:${cluster.worker.id}`;
	process.title = `reverse_proxy/${id}`;
	return new Application().run();
}

// Start the application.
main().catch(err => {
	console.error(err.message);
	process.exitCode = 1;
});
